src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -213,7 +214,24 @@ ZonedDateTime auditConversationReadTimeOfUser(Conversation conversation, User us
         ConversationParticipant readingParticipant = conversation.getConversationParticipants().stream()
                 .filter(conversationParticipant -> conversationParticipant.getUser().getId().equals(user.getId())).findAny().get();
         readingParticipant.setLastRead(ZonedDateTime.now());
-        conversationParticipantRepository.save(readingParticipant);
         return readingParticipant.getLastRead();
     }
+
+    /**
+     * Update unreadMessageCount field of ConversationParticipant
+     *
+     * @param userId            userId of the sender of the message(Post)
+     * @param courseId          courseId id of the course that the messaging is happening
+     * @param conversationId    conversationId id of the conversation with participants
+     */
+    public void unreadMessages(Long userId, Long courseId, Long conversationId) {
```

Comment:
I would suggest renaming this to something like `updateUnreadMessagesCountOfParticipants`